src/Analyzers/Core/Analyzers/PopulateSwitch/AbstractPopulateSwitchDiagnosticAnalyzer.cs

```diff
@@ -57,11 +57,14 @@ private void AnalyzeOperation(OperationAnalysisContext context)
                     .Add(PopulateSwitchStatementHelpers.MissingCases, missingCases.ToString())
                     .Add(PopulateSwitchStatementHelpers.MissingDefaultCase, missingDefaultCase.ToString());
 
+#pragma warning disable CS8620 // Mismatch in nullability of 'properties' parameter and argument types - Parameter type for 'properties' has been updated to 'ImmutableDictionary<string, string?>?' in newer version of Microsoft.CodeAnalysis (3.7.x).
                 var diagnostic = Diagnostic.Create(
                     Descriptor,
                     GetDiagnosticLocation(switchBlock),
                     properties: properties,
                     additionalLocations: new[] { switchBlock.GetLocation() });
+#pragma warning restore CS8620
```

Comment:
note certain i get what's going on. the local `properties` is non-null... so why is it a problem to pass that along.

Reply:
So, the signature of `Diagnostic.Create` was changed post 3.6.0 in https://github.com/dotnet/roslyn/pull/43023/files#diff-b3066efb626184718718c2d1642f1266R109, but the CodeStyle layer still does not reference the package version that has that change, hence the diagnostic. Once we bump up the CodeStyle layer's MS.CA version to 3.7.0, we should be able to revert this suppression.